### PR TITLE
Support for scte35-2020/PASS/Ad Tiers unit tests

### PR DESCRIPTION
**Pull request by nehashanbhag**
Unit tests and code changes for closing logic of 0x11

### CODE REVIEW

File: scte35/state.go [hunk 1]
```diff
@@ -162,13 +163,6 @@ func (s *state) ProcessDescriptor(desc SegmentationDescriptor) ([]SegmentationDe
 			err = gots.ErrSCTE35MissingOut
 			break
 		}
-		for _, d := range closed {
```
Comment (nehashanbhag) [inline scte35/state.go:165]: this closing logic doesn't seem to be what we need in SCTE35-2020. Removing it for that reason.

File: scte35/state_test.go [hunk 2]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
 		t.Errorf("There should have been 1 open signal, but we saw (%d) open signals instead.", len(state.Open()))
 	}
 }
+
+// Tests closing logic of 0x11 with 0x10, 0x20 ,0x30
```
Comment (nehashanbhag) [inline scte35/state_test.go:742]: I combined multiple open signals here to keep unit tests short. I had to write separate tests for 0x22 and 0x34,0x36,0x40 and 0x44 as the closing logic between 0x11 and these segmentation types intersect making it hard to test them in 1 test.

File: scte35/state_test.go [hunk 3]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	// 0x10
+	scteDesc10SignalBytes, _ := base64.StdEncoding.DecodeString(scteDesc10Signal)
+	outSignal, err := NewSCTE35(append([]byte{0x0}, scteDesc10SignalBytes	...))
+	if err != nil {
+		t.Fatal("Error creating SCTE-35 signal, return err:", err.Error())
+	}
+
+	closed, err := state.ProcessDescriptor(outSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
```
Comment (laba2346) [inline scte35/state_test.go:755]: thoughts on this wording?
```suggestion
		t.Errorf("ProcessDescriptor returned an error unexpectedly: %s", err.Error())
```

File: scte35/state_test.go [hunk 4]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	}
+	if len(closed) != 0 {
+		t.Errorf("No events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be one open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:761]: thoughts on this wording?
```suggestion
		t.Errorf("There should be one open signal (have %d)", len(state.Open()))
```

File: scte35/state_test.go [hunk 5]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	outSignal, err = NewSCTE35(append([]byte{0x0}, scteDesc20SignalBytes...))
+	if err != nil {
+		t.Fatal("Error creating SCTE-35 signal, return err:", err.Error())
+	}
+	closed, err = state.ProcessDescriptor(outSignal.Descriptors()[0])
+	if len(closed) != 0 {
+		t.Errorf("No events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 2 {
+		t.Errorf("There should be two open signals (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:775]: see other comment

File: scte35/state_test.go [hunk 6]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No event should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 3 {
+		t.Errorf("There should be three open signals (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:793]: see other comment

File: scte35/state_test.go [hunk 7]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 3 {
+		t.Errorf("Three events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 0 {
+		t.Errorf("There should be no open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:811]: see other comment

Comment (laba2346) [reply scte35/state_test.go:811]: also `be no open signal` -> `be no open signals`

File: scte35/state_test.go [hunk 8]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	if err != nil {
+		t.Fatal("Error creating SCTE-35 signal, return err:", err.Error())
+	}
+
+	closed, err := state.ProcessDescriptor(outSignal.Descriptors()[0])
+	if len(closed) != 0	 {
+		t.Errorf("No events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:830]: see other comment

File: scte35/state_test.go [hunk 9]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 1 {
+		t.Errorf("One event should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 0 {
+		t.Errorf("There should be no open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:848]: see other comment

File: scte35/state_test.go [hunk 10]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+
+	closed, err := state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No event should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:870]: see other comment

File: scte35/state_test.go [hunk 11]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	if err != nil {
+		t.Fatal("Error creating SCTE-35 signal, return err:", err.Error())
+	}
+
+	closed, err := state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No event should have been closed (%d were)", len(closed))
```
Comment (laba2346) [inline scte35/state_test.go:867]: `No event` -> `No events`

File: scte35/state_test.go [hunk 12]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	if err != nil {
+		t.Fatal("Error creating SCTE-35 signal, return err:", err.Error())
+	}
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No event should have been closed (%d were)", len(closed))
```
Comment (laba2346) [inline scte35/state_test.go:885]: `No event` -> `No events`

File: scte35/state_test.go [hunk 13]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	}
+	if len(state.Open()) != 2 {
+		t.Errorf("There should be 2 open signals (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:888]: see other comment

File: scte35/state_test.go [hunk 14]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	if err != nil {
+		t.Fatal("Error creating SCTE-35 signal, return err:", err.Error())
+	}
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No event should have been closed (%d were)", len(closed))
```
Comment (laba2346) [inline scte35/state_test.go:903]: `No event` -> `No events`

File: scte35/state_test.go [hunk 15]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	}
+	if len(state.Open()) != 3 {
+		t.Errorf("There should be 3 open signals (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:906]: see other comment

File: scte35/state_test.go [hunk 16]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	if err != nil {
+		t.Fatal("Error creating SCTE-35 signal, return err:", err.Error())
+	}
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No event should have been closed (%d were)", len(closed))
```
Comment (laba2346) [inline scte35/state_test.go:921]: `No event` -> `No events`

File: scte35/state_test.go [hunk 17]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+	}
+	if len(state.Open()) != 4 {
+		t.Errorf("There should be 4 open signals (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:924]: see other comment

File: scte35/state_test.go [hunk 18]
```diff
@@ -705,3 +738,207 @@ func TestOutInInOutIn36_37_10_37_37(t *testing.T) {
...
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
+	}
+	if len(closed) != 4 {
+		t.Errorf("4 events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 0 {
+		t.Errorf("There should be no open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:942]: `be no open signal` -> `be no open signals`

Comment (laba2346) [reply scte35/state_test.go:942]: see other comment as well

**Review by laba2346 — APPROVED** [review]

**Pull request merged**